### PR TITLE
WIP VSM Initial definitions of Actions, Triggers and States

### PR DESCRIPTION
**Pull request by carolhmj**

### CODE REVIEW

File: packages/tools/vsm/src/components/StateBehaviorViewComponent.tsx [hunk 1]
```diff
@@ -144,19 +144,21 @@ export const StateBehaviorViewComponent: FC = () => {
...
-        if (stateManager && graphCanvasComponent && selectedNode) {
+        if (stateManager && graphCanvasComponent && selectedNode && stateMachine) {
             // Have to clear out existing nodes.
             graphCanvasComponent.reset();
 
             const newNodes = new Array<GraphNode>();
 
             // Create nodes
-            const nodesToAdd = [
+            const nodesToAdd: any[] = [
```
Comment (RaananW) [inline packages/tools/vsm/src/components/StateBehaviorViewComponent.tsx:154]: was the type needed?

File: packages/tools/vsm/src/components/StateBehaviorViewComponent.tsx [hunk 2]
```diff
@@ -167,33 +169,37 @@ export const StateBehaviorViewComponent: FC = () => {
...
+            if (setPos && dest) {
+                newLinks.push(linkPorts(graphCanvasComponent, setPos, setPos.outputPorts[0], dest, dest.inputPorts[0]));
+            }
 
             setNodes(newNodes);
             setLinks(newLinks);
         }
     }, [stateManager, graphCanvasComponent, selectedNode]);
 
+    // @ts-ignore
```
Comment (RaananW) [inline packages/tools/vsm/src/components/StateBehaviorViewComponent.tsx:202]: is this needed?

**Review by RaananW — APPROVED** [review]
LGTM! Just 2 small comments, and a clarification for future reference - 
The behavior implementation is currently ONLY to get the current version of this pre-VSM to work. They are used to learn what is missing in core and what we can use in order to move those components into core and reuse as much a possible.

File: packages/tools/vsm/src/components/StateBehaviorViewComponent.tsx (hunk 1 above)
Comment (carolhmj) [reply packages/tools/vsm/src/components/StateBehaviorViewComponent.tsx:154]: woop not anymore, good catch! ⚽

**Pull request merged**